File: BiteClub/DTO/CreateFriendRequestBody.swift
//
//  CreateFriendRequestBody.swift
//  BiteClub
//
//  Created by Ryan Somers on 4/17/25.
//

import Foundation

struct CreateFriendRequestBody: Codable {
    let sender_id: UUID
    let receiver_id: UUID
    let status: String
}

struct FriendRequestResponse: Codable {
    let id: String
    let sender_id: String
    let receiver_id: String
    let status: String
    let created_at: String
    let updated_at: String
}

File: BiteRunr/shared/src/commonMain/kotlin/org/biterunr_studios/biterunr/DTO/CreateFriendRequestBody.kt
package org.biterunr_studios.biterunr.DTO

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class CreateFriendRequestBody(
    @SerialName("sender_id")
    val senderId: String,

    @SerialName("receiver_id")
    val receiverId: String,

    @SerialName("status")
    val status: String
)

@Serializable
data class FriendRequestResponse(
    @SerialName("id")
    val id: String,

    @SerialName("sender_id")
    val senderId: String,

    @SerialName("receiver_id")
    val receiverId: String,

    @SerialName("status")
    val status: String,

    @SerialName("created_at")
    val createdAt: String,

    @SerialName("updated_at")
    val updatedAt: String
)
